core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -525,14 +531,14 @@ private[spark] class MesosClusterScheduler(
         d.retryState.get.nextRetry.before(currentTime)
       }
 
-      scheduleTasks(
+      currentOffers = scheduleTasks(
```

Comment:
Schedule tasks returns the remaining offers after using them, after looking at this I think I can refactor it so we don't have to make it immutable.